internal/service/cleanrooms_tf/model.go

```diff
@@ -0,0 +1,608 @@
+	// and only if **asset_type** is **VIEW**
+	ViewLocalDetails []CleanRoomAssetViewLocalDetails `tfsdk:"view_local_details" tf:"optional,object"`
+	// Local details for a volume that are only available to its owner. Present
+	// if and only if **asset_type** is **VOLUME**
+	VolumeLocalDetails []CleanRoomAssetVolumeLocalDetails `tfsdk:"volume_local_details" tf:"optional,object"`
+}
+
+func (newState *CleanRoomAsset) SyncEffectiveFieldsDuringCreateOrUpdate(plan CleanRoomAsset) {
+}
+
+func (newState *CleanRoomAsset) SyncEffectiveFieldsDuringRead(existingState CleanRoomAsset) {
+}
+
+type CleanRoomAssetForeignTable struct {
+	// The metadata information of the columns in the foreign table
+	Columns catalog.ColumnInfo `tfsdk:"columns" tf:"optional"`
+}
+
+func (newState *CleanRoomAssetForeignTable) SyncEffectiveFieldsDuringCreateOrUpdate(plan CleanRoomAssetForeignTable) {
+}
+
+func (newState *CleanRoomAssetForeignTable) SyncEffectiveFieldsDuringRead(existingState CleanRoomAssetForeignTable) {
+}
+
+type CleanRoomAssetForeignTableLocalDetails struct {
+	// The fully qualified name of the foreign table in its owner's local
+	// metastore, in the format of *catalog*.*schema*.*foreign_table_name*
+	LocalName types.String `tfsdk:"local_name" tf:"optional"`
+}
+
+func (newState *CleanRoomAssetForeignTableLocalDetails) SyncEffectiveFieldsDuringCreateOrUpdate(plan CleanRoomAssetForeignTableLocalDetails) {
+}
+
+func (newState *CleanRoomAssetForeignTableLocalDetails) SyncEffectiveFieldsDuringRead(existingState CleanRoomAssetForeignTableLocalDetails) {
+}
+
+type CleanRoomAssetNotebook struct {
+	// Server generated checksum that represents the notebook version.
+	Etag types.String `tfsdk:"etag" tf:"optional"`
+	// Base 64 representation of the notebook contents. This is the same format
+	// as returned by :method:workspace/export with the format of **HTML**.
+	NotebookContent types.String `tfsdk:"notebook_content" tf:"optional"`
+}
+
+func (newState *CleanRoomAssetNotebook) SyncEffectiveFieldsDuringCreateOrUpdate(plan CleanRoomAssetNotebook) {
+}
+
+func (newState *CleanRoomAssetNotebook) SyncEffectiveFieldsDuringRead(existingState CleanRoomAssetNotebook) {
+}
+
+type CleanRoomAssetTable struct {
+	// The metadata information of the columns in the table
+	Columns catalog.ColumnInfo `tfsdk:"columns" tf:"optional"`
+}
+
+func (newState *CleanRoomAssetTable) SyncEffectiveFieldsDuringCreateOrUpdate(plan CleanRoomAssetTable) {
+}
+
+func (newState *CleanRoomAssetTable) SyncEffectiveFieldsDuringRead(existingState CleanRoomAssetTable) {
+}
+
+type CleanRoomAssetTableLocalDetails struct {
+	// The fully qualified name of the table in its owner's local metastore, in
+	// the format of *catalog*.*schema*.*table_name*
+	LocalName types.String `tfsdk:"local_name" tf:"optional"`
+	// Partition filtering specification for a shared table.
+	Partitions sharing.PartitionSpecificationPartition `tfsdk:"partitions" tf:"optional"`
```

Comment:
This should use sharing_tf instead of sharing.

Reply:
This most likely is a bug in generation, we would need to fix this before we can merge.

Reply:
@tanmay-db is it really a blocker? We don't have any CleanRoom-related resource yet...
Imho it could be fixed in the subsequent SDK updates